### ISSUE TITLE
Example 8 -- input to the CBOR representation is not clear to me

### ISSUE THREAD

**Issue by swcurran** [digitalbazaar/cel-spec#5]
I don't understand [Example 8](https://digitalbazaar.github.io/cel-spec/#example-a-minimum-viable-cryptographic-event-log) and in particular, the format of the `proofs` array?  In all the other Examples, `proof` entries are DI proofs, but in Example 8 they are just what appears to be a hash.  Perhaps some explanation is needed as to why that is?

**Comment by msporny** [comment]
> Perhaps some explanation is needed as to why that is?

Yes, that's confusing as written. The concept in play is that some CELs operate in a content-addressed system, and so those hashes would "expand" to the actual objects themselves, at which point you'd be able to have the full log, or "elide" portions of it. The concept is similar to the Gordian Envelope stuff, where you can just elide subsets of the data structure, but can still construct it if you have the objects that are related to the cryptographic hashes.

One API we're contemplating is a simple API to fetch cryptographic hashes associated with the log -- or the log provides a place where you can fetch things via cryptographic hash. 

But you're right, that part is woefully under-specified. We're struggling with understanding how much people are going to care about splitting the larger objects out of the log or not. Or if there is a log "redaction" play here or not (where you can confirm the log, but only get some of the objects in the log based on your access level. My gut tells me that this sort of stuff is too complex and, while neat, might be best avoided in v1.0.

**Comment by brianorwhatever** [comment]
> The concept in play is that some CELs operate in a content-addressed system, and so those hashes would "expand" to the actual objects themselves, at which point you'd be able to have the full log, or "elide" portions of it.

This is a really powerful concept and am in favour of exploring it further

**Comment by brianorwhatever** [comment]
In fact I think #4 is related as you're making hashes identifiers

**Comment by swcurran** [comment]
Shouldn't those be treated like URLs and include how to resolve them?  Or is it that given where the log itself is found, it is “obvious” what the proof format is and how to resolve/verify them?

**Comment by bnewbold** [comment]
I'd strongly recommend taking a look at DASL (https://dasl.ing/) as a simple set of tools for CBOR encoding objects and referencing them by hash. This is a much simplified/scoped variation of IPLD (https://ipld.io/).
